packages/db/src/meta/interface.ts

```diff
@@ -32,12 +34,29 @@ export const forAttachAndSchema = <C extends Collections>(options: {
 }) => {
   const { attach, schema } = options;
 
-  const connect = (config: TruffleConfig | ConnectOptions<C>): Db<C> => {
-    const options =
-      "working_directory" in config // TruffleConfig case
-        ? toConnectOptions(config)
-        : (config as ConnectOptions<C>);
-
+  const connect = (
+    config: TruffleConfig | ConnectOptions<C> | undefined
+  ): Db<C> => {
+    // TODO: get rid of @truffle/config stuff and rename field to "directory"
+    let options;
+    if (config && "working_directory" in config) {
+      // TruffleConfig case
+      options = toConnectOptions(config as TruffleConfig);
+    } else if (config && "directory" in config) {
+      // ConnectOptions case
+      options = config;
+    } else {
+      const configStore = new Configstore(
+        "truffle",
+        {},
+        { globalConfigPath: true }
+      );
+      const truffleDataDirectory = path.dirname(configStore.path);
+      options = {
+        directory: truffleDataDirectory,
+        adapter: (config || {}).adapter
+      };
```

Comment:
Might make sense to move this to its own module just to keep it separate/contained. How about:

```typescript
import { getDefaultDataDirectory } from "@truffle/db/meta/defaults";
```

This sounds like the sort of thing that shouldn't be defined in the meta system, and would instead be something we'd string through the `forDefinitions` functions (renamed like `forDefinitionsAndConfig`), but also seems fine to save that for later.